src/libraries/System.IO.FileSystem/src/System/IO/FileStatus.Unix.cs

```diff
@@ -65,37 +69,57 @@ internal bool IsReadOnly(ReadOnlySpan<char> path, bool continueOnError = false)
             }
 #endif
 
-            return ((_fileStatus.Mode & (int)readBit) != 0 && // has read permission
-                (_fileStatus.Mode & (int)writeBit) == 0);     // but not write permission
+            return (fileStatus.Mode & (int)readBit) != 0 && // has read permission
+                (fileStatus.Mode & (int)writeBit) == 0;     // but not write permission
         }
 
-        public FileAttributes GetAttributes(ReadOnlySpan<char> path, ReadOnlySpan<char> fileName)
+        internal static bool IsDirectory(Interop.Sys.FileStatus fileStatus)
         {
-            // IMPORTANT: Attribute logic must match the logic in FileSystemEntry
+            return (fileStatus.Mode & Interop.Sys.FileTypes.S_IFMT) == Interop.Sys.FileTypes.S_IFDIR;
```

Comment:
It is better move p/invokes from FileSystemEntry.Initialize() in the place and exclude re-reading too.
The same for all attributes.

Reply:
I talked to @Jozkee about this, we can get them moved to a separate helper class, but I would like to do that in a future PR.